.github/workflows/test.yml

```diff
@@ -3,14 +3,39 @@ on: pull_request
 env: { CLICOLOR_FORCE: 1 }
 
 jobs:
-  test:
+  cargo-fmt:
+    name: cargo-fmt@stable
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@stable
```

Comment:
This is missing a
```
with:
      components: clippy, rustfmt
```
(assuming you do merge the clippy job into this one. Otherwise drop the `clippy` part here of course.)

Reply:
Didn't realize it didn't include those components by default, I'll get that going as well.

Reply:
Github Actions has Rust including clippy and rustfmt preinstalled nowadays, so it will probably work in practice most of the time. However, that would be a bit like relying in undefined behavior.